providers/aws/provider/provider.go

```
@@ -51,18 +57,32 @@ func (s *Service) ParseCLI(req *plugin.ParseCLIReq) (*plugin.ParseCLIRes, error)
 			discoverTargets = append(discoverTargets, entry)
 		}
 	}
+
 	inventoryConfig.Discover = &inventory.Discovery{Targets: discoverTargets}
 	asset := inventory.Asset{
 		Connections: []*inventory.Config{inventoryConfig},
-		Options:     parseFlagsToOptions(flags),
+		Options:     opts,
 	}
 	return &plugin.ParseCLIRes{Asset: &asset}, nil
 }
 
+func handleAwsEc2Subcommands(args []string, opts map[string]string) *inventory.Asset {
+	asset := &inventory.Asset{}
+	switch args[1] {
+	case "instance-connect":
+		return resources.InstanceConnectAsset(args, opts)
+	case "ssm":
+		return resources.SSMConnectAsset(args, opts)
+	case "ebs":
+		return resources.EbsConnectAsset(args, opts)
+	}
```

Should we throw an error when there is something unexpected in `args[1]`?

---

Ok, just gave it a try:
```
FTL failed to run query error="cannot connect to asset, no connection info provided"
```

LGTM.